prow/plugins/trickortreat/trickortreat_test.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package trickortreat
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/github/fakegithub"
+)
+
+type fakeSnicker struct {
+	errStr string
+}
+
+func (c *fakeSnicker) readImage() (string, error) {
+	if c.errStr != "" {
+		return "", errors.New(c.errStr)
+	}
+	return fmt.Sprintf("![fake candy image](%s)", c), nil
+}
+
+func TestImages(t *testing.T) {
+	for _, imgURL := range candiesImgs {
+		t.Run(imgURL, func(t *testing.T) {
+			t.Parallel()
+			for i := 0; i < 3; i++ {
+				toobig, err := github.ImageTooBig(imgURL)
+				if err != nil {
+					t.Errorf("Failed reading image: %v", err)
+					continue
+				}
+				if toobig {
+					t.Errorf("Image %q too big", imgURL)
+				}
+				break
+			}
+		})
+	}
+}
+
+func TestReadImage(t *testing.T) {
+	if img, err := trickortreat.readImage(); err != nil {
+		t.Errorf("Could not read candies from %#v: %v", trickortreat, err)
+	} else {
```

Comment:
```golang
if err := foo(); err != nil {
  t.Errorf("whatever")
  return
}
t.Log(img)
```

is the preferred style

https://dave.cheney.net/practical-go/presentations/qcon-china.html#_return_early_rather_than_nesting_deeply

Reply:
done, img is used so changed the first line to two lines